master/internal/scheduler/resource_provider.go

```diff
@@ -1,69 +1,80 @@
 package scheduler
 
 import (
-	"bytes"
-	"fmt"
-	"strings"
-	"time"
-
-	"github.com/docker/docker/pkg/jsonmessage"
-
-	"github.com/determined-ai/determined/master/pkg/agent"
-	containerInfo "github.com/determined-ai/determined/master/pkg/container"
+	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/determined-ai/determined/master/pkg/model"
+	sproto "github.com/determined-ai/determined/master/pkg/scheduler"
 )
 
-// ContainerLog notifies the task actor that a new log message is available for the container.
-type ContainerLog struct {
-	Container containerInfo.Container
-	Timestamp time.Time
+// ResourceProvider manages the configured resource providers.
+// Currently support only one resource provider at a time.
+type ResourceProvider struct {
+	clusterID                   string
+	scheduler                   Scheduler
+	fittingMethod               SoftConstraint
+	proxy                       *actor.Ref
+	harnessPath                 string
+	taskContainerDefaults       model.TaskContainerDefaultsConfig
+	provisioner                 *actor.Ref
+	provisionerSlotsPerInstance int
 
-	PullMessage *jsonmessage.JSONMessage
-	RunMessage  *agent.RunMessage
-	AuxMessage  *string
+	resourceProvider *actor.Ref
 }
 
-func (c ContainerLog) String() string {
-	msg := ""
-	switch {
-	case c.AuxMessage != nil:
-		msg = *c.AuxMessage
-	case c.RunMessage != nil:
-		msg = strings.TrimSuffix(c.RunMessage.Value, "\n")
-	case c.PullMessage != nil:
-		buf := new(bytes.Buffer)
-		if err := c.PullMessage.Display(buf, false); err != nil {
-			msg = err.Error()
-		} else {
-			msg = buf.String()
-			// Docker disables printing the progress bar in non-terminal mode.
-			if msg == "" && c.PullMessage.Progress != nil {
-				msg = c.PullMessage.Progress.String()
-			}
-			msg = strings.TrimSpace(msg)
-		}
-	default:
-		panic("unknown log message received")
+// NewResourceProvider creates and instance of ResourceProvider.
+func NewResourceProvider(
```

Comment:
blocking: On the theme of being agnostic to particular resource providers, this should probably take an `*actor.Ref` representing the underlying resource provider and leave initialization details to the caller.